### ISSUE TITLE
Run examples in CI

### ISSUE THREAD

**Issue by rlouf** [aesara-devs/aesara#1191]
To prevent documentation for becoming stale, and make sure that users can copy/paste the example codes and run it on their machine.

**Comment by LegrandNico** [comment]
As far as I can see the documentation already use [doctest](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html) directives so building the documentation, or simply calling `make doctest` in the git workflow should work.

**Comment by rlouf** [comment]
Does it use the `.. testcode::` blocks already? If so we need to include this in the test suite asap, otherwise we need to convert as many code blocks into testcode blocks as possible.

**Comment by LegrandNico** [comment]
Yes, at least all the [scan](https://github.com/aesara-devs/aesara/blob/main/doc/library/scan.rst) examples use it already. Theano probably had this kind of CI at a certain point.

**Comment by rlouf** [comment]
Ah! Then we just have to add it to CI again.

**Comment by LegrandNico** [comment]
The [docgen.py](https://github.com/aesara-devs/aesara/blob/5043ac8e83232d7e773b80e3fc2a44cabe57016e/doc/scripts/docgen.py#L106) wrapper already implement an option to do that, you can test the documentation with:

`python ./doc/scripts/docgen.py --test`

apparently, 145 doc examples are currently falling

```
Doctest summary
===============
  756 tests
  145 failures in tests
    1 failure in setup code
    0 failures in cleanup code
build finished with problems, 65 warnings.
```

**Comment by rlouf** [comment]
I'm glad we're doing this now.

**Comment by LegrandNico** [comment]
> I'm glad we're doing this now. Many of them seem to be returning the expected result and yet the test fails. For instance:
> 
> ```
> File "library/scan.rst", line 413, in default
> Failed example:
>     location = at.imatrix("location")
>     values = at.vector("values")
>     output_model = at.matrix("output_model")
> 
>     def set_value_at_position(a_location, a_value, output_model):
>         zeros = at.zeros_like(output_model)
>         zeros_subtensor = zeros[a_location[0], a_location[1]]
>         return at.set_subtensor(zeros_subtensor, a_value)
> 
>     result, updates = aesara.scan(fn=set_value_at_position,
>                                   outputs_info=None,
>                                   sequences=[location, values],
>                                   non_sequences=output_model)
> 
>     assign_values_at_positions = aesara.function(inputs=[location, values, output_model], outputs=result)
> 
>     # test
>     test_locations = numpy.asarray([[1, 1], [2, 3]], dtype=numpy.int32)
>     test_values = numpy.asarray([42, 50], dtype=numpy.float32)
>     test_output_model = numpy.zeros((5, 5), dtype=numpy.float32)
>     print(assign_values_at_positions(test_locations, test_values, test_output_model))
> Expected:
>     [[[  0.   0.   0.   0.   0.]
>       [  0.  42.   0.   0.   0.]
>       [  0.   0.   0.   0.   0.]
>       [  0.   0.   0.   0.   0.]
>       [  0.   0.   0.   0.   0.]]
> 
>      [[  0.   0.   0.   0.   0.]
>       [  0.   0.   0.   0.   0.]
>       [  0.   0.   0.  50.   0.]
>       [  0.   0.   0.   0.   0.]
>       [  0.   0.   0.   0.   0.]]]
> Got:
>     [[[ 0.  0.  0.  0.  0.]
>       [ 0. 42.  0.  0.  0.]
>       [ 0.  0.  0.  0.  0.]
>       [ 0.  0.  0.  0.  0.]
>       [ 0.  0.  0.  0.  0.]]
> 
>      [[ 0.  0.  0.  0.  0.]
>       [ 0.  0.  0.  0.  0.]
>       [ 0.  0.  0. 50.  0.]
>       [ 0.  0.  0.  0.  0.]
>       [ 0.  0.  0.  0.  0.]]]
> ```
Yes, that's also what I was checking, the old arrays were formatted differently. Probably a bit tedious to reshape everything but I can work on that once the scan example is merged.

**Comment by rlouf** [comment]
That would be great, thank you. I just tried to paste the "Expected Output" in the doc and the error above disappeared.
